### PR TITLE
Update putSAMLInbound method

### PR DESCRIPTION
**Pull request by hwupathum**
Update putSAMLInbound method to use update method instead of delete and create

### Related Issues
- https://github.com/wso2/product-is/issues/16151

### CODE REVIEW

**Comment by hwupathum** [comment]
Need to be merged after https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/397 and framework bump

**Review by madurangasiriwardena — DISMISSED** [review]

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 1]
```diff
@@ -66,15 +66,17 @@ public static InboundAuthenticationRequestConfig putSAMLInbound(ServiceProvider
         try {
             validateSingleSignOnProfileBindings(saml2Configuration);
             if (currentIssuer != null) {
-                // Delete the current app.
                 oldSAMLSp = getSamlSsoConfigService().getServiceProvider(currentIssuer);
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:69]: Can remove rollback

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 2]
```diff
@@ -309,17 +311,17 @@ private static String base64Encode(String content) {
                 (StandardCharsets.UTF_8));
     }
 
-    private static SAMLSSOServiceProviderDTO getSamlSsoServiceProviderDTO(SAML2Configuration saml2Configuration)
-            throws APIError {
+    private static SAMLSSOServiceProviderDTO getSamlSsoServiceProviderDTO(SAML2Configuration saml2Configuration,
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:314]: change variable order

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 3]
```diff
@@ -309,17 +311,17 @@ private static String base64Encode(String content) {
...
+                                                                          boolean isUpdate) throws APIError {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:315]:
```suggestion
                                                                          boolean isUpdateOperation) throws APIError {
```

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 4]
```diff
@@ -219,43 +221,43 @@ public static void deleteSAMLServiceProvider(InboundAuthenticationRequestConfig
         }
     }
 
-    private static SAMLSSOServiceProviderDTO createSAMLSpWithManualConfiguration(SAML2ServiceProvider saml2SpModel) {
+    private static SAMLSSOServiceProviderDTO createSAMLSpWithManualConfiguration(SAML2ServiceProvider saml2SpModel,
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:224]: change method name into persist...

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 5]
```diff
@@ -66,15 +66,17 @@ public static InboundAuthenticationRequestConfig putSAMLInbound(ServiceProvider
...
-                getSamlSsoConfigService().removeServiceProvider(currentIssuer);
             }
         } catch (IdentityException e) {
             throw handleException(e);
         }
 
         try {
+            if (currentIssuer != null) {
+                // This is an update.
+                return updateSAMLInbound(application, saml2Configuration);
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:78]: Throw IdentitySAML2ClientException if false

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 6]
```diff
@@ -137,40 +139,40 @@ private static void rollbackSAMLSpRemoval(SAMLSSOServiceProviderDTO oldSAMLSp) {
     /* @Deprecated. Please use {@link #createSAMLInbound(ServiceProvider, SAML2Configuration)} instead. */
     public static InboundAuthenticationRequestConfig createSAMLInbound(SAML2Configuration saml2Configuration) {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:140]: throw IdentitySAML2ClientException in errors

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java (hunk 4 above)
Comment (hwupathum) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java]: created a separate method for update

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java [hunk 7]
```diff
@@ -171,6 +126,26 @@ public static InboundAuthenticationRequestConfig createSAMLInbound(
...
+        // Set certificate if available.
+        if (samlssoServiceProviderDTO.getCertificateContent() != null) {
+            serviceProvider.setCertificateContent(base64Encode(samlssoServiceProviderDTO.getCertificateContent()));
+        }
+
+        return returnSAMLInbound(samlssoServiceProviderDTO);
+    }
+
+    private static InboundAuthenticationRequestConfig returnSAMLInbound(
+            SAMLSSOServiceProviderDTO samlssoServiceProviderDTO) {
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java:147]: Method name should explain the functionality.

Comment (hwupathum) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java]: Changed as `createInboundAuthenticationRequestConfig`

**Pull request merged**